File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majority = nums[0], votes=1;
        for(int i=1; i<nums.size();i++)
        {
            if(votes==0)
            {
                majority = nums[i];
                votes = 1;
            }
            else if(majority==nums[i])
            {
                votes ++;
            }
            else
            {
                votes --;
            }
         
        }
        return majority;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
       
        int majority = nums[0];
        int vote = 0;
        for(int i=0;i<nums.length;i++)
        {
          if(vote==0)
          {
              majority = nums[i];
              vote = 1;
          }
          else if(majority == nums[i])
          {
              vote++;
          }
          else
          {
              vote--;
          }
        }
        return majority;
    }
}